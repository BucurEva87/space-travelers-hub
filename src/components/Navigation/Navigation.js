import React from 'react';
import { NavLink } from 'react-router-dom';
import style from './Navigation.module.scss';

const Navigation = () => {
  const activeStyle = { textDecoration: 'underline' };

  return (
    <nav className={style.nav}>
      <ul>
        <li>
          <NavLink
            to="/rockets"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/missions"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Missions
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/dragons"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Dragons
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/myprofile"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
