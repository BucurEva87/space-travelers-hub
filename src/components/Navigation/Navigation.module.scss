@import '../../styles/variables';

.nav {
  margin-left: auto;
}

.nav ul {
  display: flex;
}

.nav ul li {
  margin: 0 1rem;
}

.nav ul li:nth-child(3) {
  border-right: 0.1rem solid black;
  padding-right: 1rem;
}

.nav ul li:nth-child(4) {
  margin-left: 0;
}

.nav ul li a {
  font-size: 0.8rem;
  color: $linkBlue;
}
