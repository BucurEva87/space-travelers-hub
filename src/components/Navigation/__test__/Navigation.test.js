import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../Navigation';

describe('Navigation component', () => {
  test('renders on the screen', () => {
    expect(render(
      <Router>
        <Navigation />
      </Router>,
    ).getByTestId('navbar')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Router>
        <Navigation />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
