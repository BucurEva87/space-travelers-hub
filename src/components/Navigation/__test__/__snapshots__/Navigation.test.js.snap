// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<nav
  className="nav"
  data-testid="navbar"
>
  <ul>
    <li>
      <a
        className=""
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className=""
        href="/dragons"
        onClick={[Function]}
      >
        Dragons
      </a>
    </li>
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
</nav>
`;
